pkg/agent/nplagent/rules/rules.go

```diff
@@ -0,0 +1,57 @@
+// +build !windows
+
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rules
+
+import (
+	"github.com/coreos/go-iptables/iptables"
+	nplutils "github.com/vmware-tanzu/antrea/pkg/agent/nplagent/lib"
+)
+
+type PodPortRules interface {
+	Init() (bool, error)
+	AddRule(port int, podip string) (bool, error)
+	DeleteRule(port int, podip string) (bool, error)
+	SyncState(podPort map[int]string) bool
+	GetAllRules(podPort map[int]string) (bool, error)
+	DeleteAllRules() (bool, error)
```

Comment:
I don't understand why most of these interface functions return both a boolean (to indicate success) and an error, instead of just an error?